### PR TITLE
Add basic toast message support

### LINKED ISSUE #258: Add toast messages as feedback to the user
**Is your feature request related to a problem? Please describe.**
Feedback from the app or from the server does currently not display in a good way for the user.

**Describe the solution you'd like**
The messages should be displayed to the user.
This can easily be done by using something like 'toast'-messages.

Toast messages are small messages that appear on the page to notify users that something has happened.
They usually appear at the edge of the screen and will last for a set number of seconds before they disappear.

Colors or accent colors can be used to display more context. Like red for warnings and green for success.

**Additional context**
An example of usage could be event attendance. Where the result of you trying to attend the event is displayed.

Another would be paying for an event or adding money to Nibble saldo.

### PR DESCRIPTION
**Pull request by oleast**
Adds toast messages to with a few usable settings to make it easy to display messages to the user when something is happening.

resolves #258 

_Disclaimer: The stylish button on the top are not included in this PR_
new-toast

### CODE REVIEW

**Comment by oleast** [comment]
Compared to the gif above, the countdown now ticks in 200 ms intervals instead of a second.

File: src/common/hooks/useCountDown.ts [hunk 1]
```diff
@@ -3,22 +3,25 @@ import { useEffect } from 'react';
 
 import { useDecrement } from './useDecrement';
 
-const calculateOffset = (time: DateTime): number => {
+const DEFULAT_TICK = 1000;
```
Comment (mariush2) [inline src/common/hooks/useCountDown.ts]: Spelling error, shold be "default"

File: src/core/utils/toast/toast.less [hunk 2]
```diff
@@ -0,0 +1,54 @@
+@import '~common/less/mixins.less';
+@import '~common/less/constants.less';
+@import '~common/less/colors.less';
+
+.toastContainer {
+  position: fixed;
+  bottom: 0;
+  left: 0;
+  box-sizing: border-box;
+  border: 8px solid transparent;
```
Comment (mariush2) [inline src/core/utils/toast/toast.less]: The toast-messages should have a border-radius of 3px (same as .owCard())

File: src/core/utils/toast/useToast.ts [hunk 3]
```diff
@@ -0,0 +1,7 @@
+import { useContext } from 'react';
+
+import { IToastContextState, ToastContext } from './ToastContext';
+
+export const useToast = (): IToastContextState => {
```
Comment (mariush2) [inline src/core/utils/toast/useToast.ts]: Burde denne være 'export default'?

Comment (oleast) [reply src/core/utils/toast/useToast.ts]: I generally prefer not using default exports.
This enforces the name of the exported function and generally makes things easier to import automatically with with the editor.

**Comment by oleast** [comment]
> Great work! The only thing I'm abit unsure the colours on the messages though. I think it would make more sense to have the card be white with black text. This would also make the coloured text look better aswell.

The problem with having white cards is that they blend completly in with the rest of the content on the page. They overlap in a strange way, making it seem like the page is broken.

This is a general problem that can be added to the list in #232 

I don't really know how to solve this well.

File: src/core/utils/toast/toast.less (hunk 2 above)
Comment (oleast) [reply src/core/utils/toast/toast.less]: This border is not visible. It is used as used as the margin around the entire list of messages.

The reason for not using a margin is that the content would overflow inside the container as it is set to a width of `100vw`, which a margin would not uphold.

**Comment by oleast** [comment]
I worked a bit on making the toasts a little more in-tune with the current design.
**The GIF in the first post has been updated to reflect this**

They now use the `owCard` mixin, and uses the `darkGray` background.
Messages are now separated by a little bap between them.
The text and progress bar has also been made bold.

Colors have been turned off for now, as they are not immediately needed, and most messages work without them.

The `useToast` hook has been updated with settings. These will let you:
- Set the hook to remove the previous message created _by the same hook_ before adding a new message. (as displayed above).
- Setting the default duration and type of messages created by that hooks `addMessage` function.

**Pull request merged**